### PR TITLE
DSP-18805: Native is now the default engine when creating a graph

### PR DESCRIPTION
**Pull request by nastra**

### CODE REVIEW

**Comment by rocco408** [comment]
@nastra  do you plan to update this too? https://github.com/riptano/graph-examples/blob/master/fraud/pom.xml#L8

If so, then I _think_ we need the dse-spark-dependencies published for 6.8, I asked about this in our doc about this https://docs.google.com/document/d/15j_QvHqCchRjF82gv4Ohzfq689QmVRSWwNUWS5sERhw/edit?ts=5cc88402#heading=h.yhryvyx5etnd

**Comment by xytxytxyt** [comment]
sadly it won't work with 6.0 and 6.7, but i don't know how to fix it...

maybe some bash version string parsing in https://github.com/riptano/haxx/blob/master/templates/graph/dsegf-statestreet-dataloading-template.yaml.jinja2#L89-L97 ?

**Comment by xytxytxyt** [comment]
or if we could get the dse version in groovy in dse gremlin-console we could do it right in the schema

**Comment by nastra** [comment]
@rocco408 it seems that the spark depedencies definitely need to be published for 6.8 as the project doesn't compile without those (with/without my changes)

File: fraud/statestreet-legacy-schema.groovy [hunk 1]
```diff
@@ -1,5 +1,5 @@
 // set RF=1 due to limited space
-system.graph('statestreet').replication("{'class' : 'NetworkTopologyStrategy', 'SearchGraphAnalytics' : 1}").systemReplication("{'class' : 'NetworkTopologyStrategy', 'SearchGraphAnalytics' : 1}").create()
+system.graph('statestreet').replication("{'class' : 'NetworkTopologyStrategy', 'SearchGraphAnalytics' : 1}").systemReplication("{'class' : 'NetworkTopologyStrategy', 'SearchGraphAnalytics' : 1}").engine(Classic).create()
```
Comment (xytxytxyt) [inline fraud/statestreet-legacy-schema.groovy:2]: is it possible to catch the error here and retry without `engine(Classic)` to make it work with 6.0 and 6.7?

Comment (nastra) [reply fraud/statestreet-legacy-schema.groovy:2]: do you already have specific code to do that? Otherwise I'll have to look into that on monday

**Comment by rocco408** [comment]
+1 on the dse-spark-dependencies requirement. I've pinged Brian Cantoni in the linked Google Doc about this. I'll follow-up on this.

File: fraud/statestreet-legacy-schema.groovy (hunk 1 above)
Comment (xytxytxyt) [reply fraud/statestreet-legacy-schema.groovy:2]: monday is okay, probably no tests will work over the weekend anyway :)

**Comment by xytxytxyt** [comment]
i tried it out, this works:

6.7.4:
```
gremlin> try {
......1>     Classic
......2>     system.graph('my_classic_graph').engine(Classic).create()
......3> } catch (MissingPropertyException mpe) {
......4>     system.graph('my_classic_graph').create()
......5> }
==>null
gremlin> system.graphs()
==>my_classic_graph
```

6.8.0:
```
gremlin> try {
......1>     Classic
......2>     system.graph('my_classic_graph').engine(Classic).create()
......3> } catch (MissingPropertyException mpe) {
......4>     system.graph('my_classic_graph').create()
......5> }
==>You have created the graph 'my_classic_graph' with Classic engine.
If you wanted to create a graph using the newer Native engine, you should create your graph as follows:
system.graph('my_classic_graph').withReplication("{'class' : 'NetworkTopologyStrategy', '<DC_NAME>' : <RF> }").create()


OK
gremlin> system.graphs()
==>my_classic_graph
```

**Comment by xytxytxyt** [comment]
not sure if we want to rename `statestreet-legacy-schema.groovy` to classic considering haxx is not ready for it yet...

**Comment by nastra** [comment]
we'll implement https://datastax.jira.com/browse/DSP-18829 and won't have to do any version checks

File: fraud/statestreet-legacy-schema.groovy (hunk 1 above)
Comment (nastra) [reply fraud/statestreet-legacy-schema.groovy:2]: https://datastax.jira.com/browse/DSP-18829 will be the solution

**Pull request merged**